### DISCUSSION TITLE
How to add points to 2 differents series on the same graph?

### DISCUSSION THREAD

**Discussion by SomeUsername14786** [JuliaPlots/Makie.jl#3692] [Q&A] [answered]
Hi,

I wish to make a line graph with 2 differents series:

```
lines(1:4, rand(4)
lines!(1:4, rand(4)
```

So far so good.

Now I want to add additional points to each series as a calculation progress similar to what we can see on this example of Agents.jl: https://juliadynamics.github.io/Agents.jl/stable/examples/agents_visualizations/.

I tried something like that to initialize the graph:

```
using GLMakie, Makie

y1 = Observable([60.0, 65.3, 45.5])
y2 = Observable([30.0, 25.7, 98.9])

x1 = lift(y1) do y1
    1:length(y1)
end

x2 = lift(y2) do y2
    1:length(y2)
end

lines(x1, y1)
lines!(x2, y2)
ylims!(0, 100)
```

And to update the graph:
```
newValue = 55.0
append!(y1[], newValue)
```

However neither the graph nor the `x1` observable are updated when I update the `y1` observable.

What am I doing wrong? What's the correct way to do it?

**Comment by asinghvi17** [comment, marked as answer]
That append only accesses the observable's value and doesn't do anything to it.  You can `notify(y1)` afterwards to let the observable know it's been updated. 

If you were assigning to the variable as `y1[] = ...`, then there would be no need to `notify`.

Also, have you seen the `timeseries` recipe?

**Comment by SomeUsername14786** [comment]
Thanks a lot! Using `notify(y1)` solved the problem. I also needed to manually call `xlims!` otherwise the graph doesn't expand by itself.

What are `timeseries`? Only documentation I found was this one: https://docs.makie.org/stable/api/#TimeSeries which is rather short.
